### PR TITLE
Support gen::Module::get_submodule/_mut() methods

### LINKED ISSUE #48: Allow `get` for `Module`
When building up a `Module` and adding a few submodules, it would be handy to be able to introspect what I have in there. 

Probably should have them as a Map, instead of a vec, and then expose the normal `get/_mut()` methods.

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #48

### CODE REVIEW

File: src/gen/module.rs [hunk 1]
```diff
@@ -105,9 +106,17 @@ impl Module {
     }
     /// Add submodule
     pub fn add_submodule(&mut self, module: Module) -> &mut Self {
-        self.sub_modules.push(module);
+        self.sub_modules.insert(module.name.clone(), module);
         self
     }
+    /// Get a mutable reference to a submodule of this module
+    pub fn get_submodule_mut(&mut self, name: impl ToString) -> Option<&mut Module> {
```
Comment (DarkDrek) [inline src/gen/module.rs]: Can we use the same bound as https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get `Borrow<T>` for these methods?

Comment (milesgranger) [reply src/gen/module.rs]: I must be doing something silly.
```rust
pub fn get_submodule_mut<Q: ?Sized, K>(&mut self, name: &Q) -> Option<&mut Module>
        where
            K: Borrow<Q>,
            Q: Hash + Eq,
            String: Borrow<Q>
```

Results in 
```
assert!(module.get_submodule_mut("Bar2").is_none());
   |                    ^^^^^^^^^^^^^^^^^ cannot infer type for `K`

```

Comment (DarkDrek) [reply src/gen/module.rs]: The not mut version should look like this
```rust 
pub fn get_submodule<Q>(&self, name: &Q) -> Option<&String>
    where
        String: Borrow<Q>,
        Q: ?Sized + Hash + Eq,

Comment (milesgranger) [reply src/gen/module.rs]: Sweet, thank you, sir!

**Review by DarkDrek — APPROVED** [review]

**Pull request merged**